ui/lib/utils/sentryHelpers.ts

```diff
@@ -0,0 +1,65 @@
+import * as Sentry from '@sentry/react'
+
+const firstMountTransaction = Sentry.startTransaction({ name: 'first mount' })
+const transactionHub = new Map<string, typeof firstMountTransaction>()
+
+export const sentryEnabled = process.env.REACT_APP_SENTRY_ENABLED === 'true'
+
+export function markStart(name: string) {
+  const transaction = Sentry.startTransaction({ name })
+  transactionHub.set(transaction.name, transaction)
+  transactionHub.set(transaction.traceId, transaction)
+  return transaction
+}
+
+export function markEnd(name: string, traceId?: string) {
+  const transaction = traceId
+    ? transactionHub.get(traceId)
+    : transactionHub.get(name)
+  if (transaction) {
+    transaction.finish()
+    transactionHub.delete(name)
+    if (traceId) {
+      transactionHub.delete(traceId)
+    }
+  }
+}
+
+export function markTag(key: string, value: string | number, traceId: string) {
+  const transaction = transactionHub.get(traceId)
+  if (transaction) {
+    transaction.setTag(key, value)
+  }
+}
+
+export const reportError: typeof Sentry.captureException = (...args) => {
+  if (sentryEnabled) {
+    return Sentry.captureException(...args)
+  }
+  return ''
+}
+
+export function initSentryRoutingInstrument() {
+  window.addEventListener('single-spa:first-mount', () => {
+    firstMountTransaction.finish()
+  })
+
+  window.addEventListener('single-spa:before-routing-event', (e: any) => {
+    const transaction = markStart('routing')
+    transaction.setData('routing.detail', e.detail)
+    transaction.setTag('routing.from', e.detail.oldUrl)
+    transaction.setTag('routing.to', e.detail.newUrl)
```

Comment:
Same as the mixpanel, we can't collect the URL prefix and some sensitive information.

Reply:
fixed, now ot only collect the pathname